### ISSUE TITLE
[BUG] ? OnInitializedAsync is not called for modules on the next url if it exist on the calling page

### ISSUE THREAD

**Issue by beolafsen** [oqtane/oqtane.framework#4391]
### Oqtane Info

Version - Dev branch (5.2.0)  - downloaded 1 july-24.
Render Mode - Interactive
Interactivity - Server
Database - SQL Server

### Describe the bug
OnInitializedAsync is not called for modules on the next url if the module exist on the calling page - it only show the moduledata as it was on the calling page.

Ex: 
I have two testpages. Both contain the same module. The module contain only a combo box with data. The data filled in the combox is defined in the Module Settings. It could be tournaments, tournamentclasses, membertypes, semesters, etc.

The module on the first page is setup to show tournaments 
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/55cca613-89d8-4cc0-888c-dc164584e44a)

and on the secound page the module is setup to show tournamnetclasses.
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/753a468a-09bf-49e0-a8c0-4c6179871b64)

If I goto one of this pages from a page that don't contain the combobox module - the OnInitializedAsync  is called ant it is filled with the correct data.

If I go from a page that contain this Module  to another page that contain the same Module it show the data in the combobox from the calling url and the OnInitializedAsync is not called in the Module in the new page.
So the first page looks like this (which is correct):
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/f1d316a5-9531-42b3-b46a-1b435cf184de)
and then I go to the second page that should contain tournamentclasses - but it contain the same data as the first (calling) page - and the OnInitializedAsync is not called:
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/7531aaea-2dc0-4402-93a1-af8d8ca8c612)
After click the pencil icon (
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/2fc39759-3bf4-4c6d-a04b-258c9814cb2e)
) the OnInitializedAsync is called and the module change the data to tournamentclasses:
![image](https://github.com/oqtane/oqtane.framework/assets/76835718/62d3aa6a-0b4a-4952-8a36-6d044fdf6b37)

### Expected Behavior
OnInitializedAsync  are called for all Modules when we go to a new page.

**Comment by sbwalker** [comment]
In Interactive render mode, Blazor uses a single page application (SPA) approach. There are no "pages" - only a single page where the URL changes and the router determines which components should be rendered. If a component is already on the page then OnInitialized will not fire when the URL changes - you need to use the OnParametersSet instead.

**Comment by beolafsen** [comment]
Aha...Thank you for helping.

**Comment by beolafsen** [comment]
Hello! 
I have a problem that I can't figure out when it comes to handling properties events that are sent between modules. I think it must have something to do with how SinglePage works.

In almost all my modules, I have a selection of datasets. E.g. Before I show prices, you must specify for which Semester the prices should be shown for. If you want to show invoices, you must first specify for which customer, etc.

Since I use the same type of selection in many modules, I have chosen to put the selection in a separate module that handles all types of selections, and every time you change a selection, an Event is fired. Basicaly it looks like this and how I want it to work.
![image](https://github.com/user-attachments/assets/66b591ed-05eb-4d42-abe1-85555c4a5cce)
and this is how it looks on screen:
![image](https://github.com/user-attachments/assets/7fd52399-61ea-45a1-b0e9-3af863a12fea)

So - The problem:
Every time I make a new selection and I fire of an Event - All the modules that has been opened get this event and start to work with the event. I only want the module I have on the screen to work with the event - not all the moduls that has been opened since I started my web-site.

![image](https://github.com/user-attachments/assets/166ea25c-7b91-4352-bdcd-185fb458807b)

In my example I open first the modules on PageId1, then PageId2 and now I have PageId3 on my screen.
Then I select a new Semester in my SelectModule and this is sent to Module4 (that is listen for MySemesterId) - but ModuleId2 and ModuleId3 also get this event and start to work with it. MdouleId1 is also listen for MySemesterId and will start to work with that (read new data etc) - but that is waist of time - because there are nobody that will see this data.

I have implemented the :
    **public void Dispose() => ((INotifyPropertyChanged)SiteState.Properties).PropertyChanged -= PropertyChanged;**
on every page which I thought would make Akutell Modul stop listening for events when I leave it.

Hope this is understandable. 
Have I solved this in a way (with 2 modules talking to each other) that Oqtane/.net is unable to handle. Do I need to change the selection to become part of the module - e.g. by creating a component?

I use the following settings: 
![image](https://github.com/user-attachments/assets/53af195d-2acc-422a-aab6-1a5fa4869e3f)
